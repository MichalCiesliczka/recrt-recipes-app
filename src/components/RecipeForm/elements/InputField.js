// @flow
import React from 'react';
import type { FieldProps } from 'redux-form';

import TextField from '@material-ui/core/TextField';
import FormHelperText from '@material-ui/core/FormHelperText';

type Props = {
  label: string,
  multiline: boolean,
} & FieldProps;

const InputField = ({
  input,
  label,
  multiline,
  meta: {
    touched,
    error,
  },
  ...custom
}: Props) => (
  <div className="input-wrapper">
    <TextField
      fullWidth
      label={label}
      multiline={multiline}
      error={touched && !!error}
      {...input}
      {...custom}
    />
    {
      touched && error && (
        <FormHelperText error>
          {error}
        </FormHelperText>
      )
    }
  </div>
);

export default InputField;
