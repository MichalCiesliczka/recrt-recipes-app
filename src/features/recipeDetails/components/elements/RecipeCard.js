// @flow
import React from 'react';
import type { Node } from 'react';

import { withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

type Props = {
  title: string,
  classes: {
    title: string,
  },
  children: Node,
};

const styles = {
  title: {
    fontWeight: 'bold',
    fontSize: 18,
  },
};

const RecipeCard = ({ title, children, classes }: Props) => (
  <Card>
    <CardContent>
      <Typography
        component="p"
        className={classes.title}
      >
        {title}
      </Typography>
      {children}
    </CardContent>
  </Card>
);

export default withStyles(styles)(RecipeCard);
