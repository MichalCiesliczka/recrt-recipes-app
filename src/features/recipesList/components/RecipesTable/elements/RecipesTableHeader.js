import React from 'react';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';

import type { RecipesListColumns } from '../../../recipesList.types';

type Props = {
  columns: RecipesListColumns,
};

const RecipesTableHeader = ({ columns }: Props) => (
  <TableRow>
    <TableCell>
      No.
    </TableCell>
    {
      columns.map(column => (
        <TableCell
          key={column.value}
        >
          {column.label}
        </TableCell>
      ))
    }
  </TableRow>
);

export default RecipesTableHeader;
