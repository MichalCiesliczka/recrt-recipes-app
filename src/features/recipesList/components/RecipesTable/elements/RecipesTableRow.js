// @flow
import React from 'react';
import { compose } from 'redux';
import { withRouter } from 'react-router';

import { withStyles } from '@material-ui/core/styles';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';

import type { RecipesListColumns } from '../../../recipesList.types';
import type { Recipe } from '../../../../recipeDetails/recipeDetails.types';

// TODO: Add type for history React-Router object.
type Props = {
  recipe: Recipe,
  index: number,
  columns: RecipesListColumns,
  history: {
    push: (url: string) => void,
  },
  classes: {
    column: string,
    'column--big': string,
  },
};

const styles = {
  column: {
    maxWidth: 300,
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    whiteSpace: 'nowrap',
  },
  'column--big': {
    maxWidth: 500,
  },
};

export const RecipesTableRow = ({
  recipe,
  index,
  columns,
  history,
  classes,
}: Props) => {
  const redirectToDetails = id => history.push(`/recipe/${id}`);
  return (
    <TableRow
      hover
      onClick={() => redirectToDetails(recipe.id)}
      style={{
        cursor: 'pointer',
      }}
    >
      <TableCell>{index + 1}</TableCell>
      {
        columns.map(column => (
          <TableCell
            key={column.value}
            className={`${classes.column} ${column.isBig ? classes['column--big'] : ''}`}
            big={column.isBig ? column.isBig.toString() : undefined}
          >
            {recipe[column.value]}
          </TableCell>
        ))
      }
    </TableRow>
  );
};

export default compose(
  withRouter,
  withStyles(styles),
)(RecipesTableRow);
