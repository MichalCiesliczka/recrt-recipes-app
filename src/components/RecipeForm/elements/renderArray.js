// @flow
import React from 'react';
import { Field } from 'redux-form';
import type { FieldArrayProps } from 'redux-form';

import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import FormHelperText from '@material-ui/core/FormHelperText';

import InputField from './InputField';

type Props = {
  title: string,
  buttonLabel: string,
  elements: string[],
  classes: {
    container: string,
    list: string,
    removeButton: string,
  },
} & FieldArrayProps;

const styles = {
  container: {
    margin: '20px 0',
  },
  list: {
    position: 'relative',
    margin: 30,
    paddingTop: 20,
  },
  removeButton: {
    position: 'absolute',
    right: 0,
    top: 0,
  },
};

const renderArray = ({
  title,
  buttonLabel,
  fields,
  elements,
  classes,
  meta,
}: Props) => (
  <div className={classes.container}>
    <Typography component="p">
      {title}
    </Typography>
    {
      meta.submitFailed && meta.error && (
        <FormHelperText error>
          {meta.error}
        </FormHelperText>
      )
    }
    <Button
      type="button"
      variant="contained"
      className="add-button"
      onClick={() => fields.push({})}
    >
      {buttonLabel}
    </Button>
    {
      fields.map((field, index) => (
        /* eslint-disable react/no-array-index-key */
        <div key={index} className={classes.list}>
          <Button
            type="button"
            className={classes.removeButton}
            onClick={() => fields.remove(index)}
            color="secondary"
            variant="fab"
            mini
          >
            X
          </Button>
          {
            elements.map(element => (
              <Field
                key={element}
                name={`${field}.${element}`}
                component={InputField}
                label={element}
              />
            ))
          }
        </div>
      ))
    }
  </div>
);

export default withStyles(styles)(renderArray);
