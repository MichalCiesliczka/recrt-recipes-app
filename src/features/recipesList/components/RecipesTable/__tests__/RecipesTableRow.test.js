import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { RecipesTableRow } from '../elements/RecipesTableRow';

describe('<RecipesTableRow />', () => {
  const defaultProps = {
    recipe: {
      id: '1',
      title: 'Test Title',
      description: 'Test Description',
      steps: [
        { name: 'Step 1' },
      ],
      ingredients: [
        { amount: 2, name: 'ingr' },
      ],
    },
    index: 1,
    columns: [
      {
        label: 'Test label',
        value: 'title',
        isBig: false,
      },
      {
        label: 'Test label 2',
        value: 'description',
        isBig: true,
      },
    ],
    classes: {
      column: 'test class',
      'column--big': 'test big class',
    },
    history: {
      push: jest.fn(),
    },
  };

  it('renders without crashing', () => {
    shallow(<RecipesTableRow {...defaultProps} />);
  });

  it('should render default status', () => {
    const tree = renderer
      .create(<RecipesTableRow {...defaultProps} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should redirect to recipe details on click', () => {
    const wrapper = shallow(<RecipesTableRow {...defaultProps} />);

    wrapper.simulate('click');

    expect(defaultProps.history.push.mock.calls.length).toEqual(1);
    expect(defaultProps.history.push.mock.calls[0][0]).toEqual('/recipe/1');
  });
});
