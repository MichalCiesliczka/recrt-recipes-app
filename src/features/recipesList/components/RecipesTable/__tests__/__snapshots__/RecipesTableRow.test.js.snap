// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipesTableRow /> should render default status 1`] = `
<tr
  className="MuiTableRow-root-1 MuiTableRow-hover-3"
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  <td
    aria-sort={null}
    className="MuiTableCell-root-6"
    scope={undefined}
  >
    2
  </td>
  <td
    aria-sort={null}
    big={undefined}
    className="MuiTableCell-root-6 test class "
    scope={undefined}
  >
    Test Title
  </td>
  <td
    aria-sort={null}
    big="true"
    className="MuiTableCell-root-6 test class test big class"
    scope={undefined}
  >
    Test Description
  </td>
</tr>
`;
